src/vm/environment.ts

```diff
@@ -103,7 +101,15 @@ export class VMServiceEnvironment extends JDServiceClient {
     }
 
     public async refreshRegistersAsync() {
-        await refreshRegistersAsync(this._registers)
+        for (const k in this._registers) {
+            const register = this._registers[k]
+            let retry = 0
+            let val: any = undefined
+            do {
+                await register.refresh()
```

Comment:
refresh does retries already

Reply:
you wrote this code for me. Are you sure?